File: MTGSearch/src/test/java/com/dbottillo/mtgsearchfree/model/MTGSetTest.java
package com.dbottillo.mtgsearchfree.model;

import android.os.Parcel;
import android.test.suitebuilder.annotation.SmallTest;

import com.dbottillo.mtgsearchfree.BaseTest;

import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

@SmallTest
public class MTGSetTest extends BaseTest {

    MTGSet set;

    @Before
    public void setup() {
        set = new MTGSet(100);
        set.setName("Zendikar");
        set.setCode("ZEN");
    }

    @Test
    public void mtgSet_ParcelableWriteRead() {
        Parcel parcel = Parcel.obtain();
        set.writeToParcel(parcel, set.describeContents());
        parcel.setDataPosition(0);

        MTGSet createdFromParcel = MTGSet.CREATOR.createFromParcel(parcel);
        assertThat(createdFromParcel, is(set));
    }
}

File: MTGSearch/src/androidTest/kotlin/com/dbottillo/mtgsearchfree/model/MTGSetTest.kt
package com.dbottillo.mtgsearchfree.model

import android.os.Parcel
import android.support.test.runner.AndroidJUnit4
import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.MatcherAssert.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MTGSetTest {

    lateinit var set: MTGSet

    @Before
    fun setup() {
        set = MTGSet(100)
        set.setName("Zendikar")
        set.setCode("ZEN")
    }

    @Test
    fun mtgSet_ParcelableWriteRead() {
        val parcel = Parcel.obtain()
        set.writeToParcel(parcel, set.describeContents())
        parcel.setDataPosition(0)

        val createdFromParcel = MTGSet.CREATOR.createFromParcel(parcel)
        assertThat(createdFromParcel, `is`(set))
    }
}